examples/force_control/Force_control_example.py

```diff
@@ -0,0 +1,236 @@
+        sphere.position = np.array([0.36, 0, 0.8-0.0005*t])
+
+        # get current end-effector position and velocity in the task/operational space
+        x = robot.get_link_world_positions(link_id)
+        dx = robot.get_link_world_linear_velocities(link_id)
+        o = robot.get_link_world_orientations(link_id)
+        do = robot.get_link_world_angular_velocities(link_id)
+
+        # Get joint positions
+        q = robot.get_joint_positions()
+
+        # Get linear jacobian
+        if robot.has_floating_base():
+            J = robot.get_jacobian(link_id, q=q)[:, qIdx + 6]
+        else:
+            J = robot.get_jacobian(link_id, q=q)[:, qIdx]
+
+        # Pseudo-inverse: \hat{J} = J^T (JJ^T + k^2 I)^{-1}
+        Jp = robot.get_damped_least_squares_inverse(J, damping)
+
+        dv = kp * (sphere.position - x) - kd * dx
+        dw = kp * quaternion_error(sphere.orientation, o) - kd * do
+        # evaluate damped-least-squares IK
+        dq = Jp.dot(np.hstack((dv, dw)))
+
+        # set joint positions
+        # robot.set_joint_velocities(dq, joint_ids=joint_ids)
+        q = q[qIdx] + dq * dt
+        robot.set_joint_positions(q, joint_ids=joint_ids)
+
+        if FT_sensor.sense() is not None:
+            if FT_sensor.sense()[2] > Fz_desired:
+                break
+        # step in simulation
+        world.step(sleep_dt=dt)
+    sp_z = []
+    num = []
+    detx = np.array([0.0, 0.0, 0.0])
+    for t in count():
+        Fz_desired = 100
+        # move sphere
+        if t == 0:
+            z = robot.get_link_world_positions(link_id)[2]
+            sphere.position = np.array([0.46 - r * np.sin(w * t + np.pi / 2), r * np.cos(w * t + np.pi / 2), z])
+
+        # get current end-effector position and velocity in the task/operational space
+        x = robot.get_link_world_positions(link_id)
+        dx = robot.get_link_world_linear_velocities(link_id)
+        o = robot.get_link_world_orientations(link_id)
+        do = robot.get_link_world_angular_velocities(link_id)
+
+        # Get joint positions
+        q = robot.get_joint_positions()
+
+        # Get linear jacobian
+        if robot.has_floating_base():
+            J = robot.get_jacobian(link_id, q=q)[:, qIdx + 6]
+        else:
+            J = robot.get_jacobian(link_id, q=q)[:, qIdx]
+
+        # Pseudo-inverse: \hat{J} = J^T (JJ^T + k^2 I)^{-1}
+        Jp = robot.get_damped_least_squares_inverse(J, damping)
+        # Apply the admittance control
+        Fz_current = FT_sensor.sense()[2]  # record the current Fz
+
+        Fz_error = Fz_current - Fz_desired  # record the current error
+
+        # set the M\D\K parameters by heuristic method, these parameters may have a good result
+        M = 1
+        D = 9500
+        K = 500000
+        # Refer the formula in this article
+        # [1] SONG, Peng; YU, Yueqing; ZHANG, Xuping. A tutorial survey and comparison of impedance control on robotic manipulation. Robotica, 2019, 37.5: 801-836.
```

Comment:
You can move the reference to the code description at the beginning of the file and here you only need to refer to [1]